### PR TITLE
test: Code Coverage: Add Edit Mileage #3 Part 3

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 77010c5</samp>

This pull request updates the test files and mock data for the add-edit-mileage page to test the autofill feature for mileage expenses. It modifies the `add-edit-mileage-2.spec.ts` file to use new mock data and add new test cases. It also adds new mock data objects for org settings and unflattened mileage transactions in `org-settings.data.ts` and `unflattened-txn.data.ts` files. Additionally, it mocks the `getDefaultVehicleType` method in the `add-edit-mileage.page.setup.spec.ts` file.

### <samp>🤖 Generated by Copilot at 77010c5</samp>

> _Mock data for tests_
> _`getNewExpense` checked well_
> _Autofill in spring_

### <samp>🤖 Generated by Copilot at 77010c5</samp>

*  Add mock data for org settings with autofill feature and unflattened mileage transactions ([link](https://github.com/fylein/fyle-mobile-app/pull/2287/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951R1288-R1295), [link](https://github.com/fylein/fyle-mobile-app/pull/2287/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfR2750-R2811))
*  Update import statements and remove unused variables and imports in `add-edit-mileage-2.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2287/files?diff=unified&w=0#diff-035f26539504516a07a4540ded16b687e8c0447825f17da32e9cfc9c47c9e3f5L2-R17), [link](https://github.com/fylein/fyle-mobile-app/pull/2287/files?diff=unified&w=0#diff-035f26539504516a07a4540ded16b687e8c0447825f17da32e9cfc9c47c9e3f5L45-R55))
*  Add test cases for `getNewExpense` method with and without autofill feature in `add-edit-mileage-2.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2287/files?diff=unified&w=0#diff-035f26539504516a07a4540ded16b687e8c0447825f17da32e9cfc9c47c9e3f5R336-R426))
*  Add `getDefaultVehicleType` method to transactionService spy object in `add-edit-mileage.page.setup.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2287/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR111))

## Clickup
https://app.clickup.com/t/85ztjmtf6

## Code Coverage
`32.97% Statements 278/843`
`28.38% Branches 195/687`
`28.91% Functions 85/294`
`33.37% Lines 268/803`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-mileage/add-edit-mileage-2.spec.ts [hunk 1]
```diff
@@ -330,5 +333,96 @@ export function TestCases2(getTestBed) {
...
+
+    describe('getNewExpense():', () => {
+      beforeEach(function () {
+        jasmine.clock().install();
+      });
+
+      it('should get a new expense object', (done) => {
+        const date = new Date('2023-08-21T07:43:15.592Z');
+        jasmine.clock().mockDate(date);
+        transactionService.getDefaultVehicleType.and.returnValue(of('CAR'));
```
Comment (Dimple16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-2.spec.ts:345]: `'CAR'` shouldn't this be an enum?

File: src/app/fyle/add-edit-mileage/add-edit-mileage-2.spec.ts [hunk 2]
```diff
@@ -330,5 +333,96 @@ export function TestCases2(getTestBed) {
...
+
+        component.getNewExpense().subscribe((res) => {
+          expect(res).toEqual(newExpenseMileageData1);
+          expect(transactionService.getDefaultVehicleType).toHaveBeenCalledTimes(1);
+          expect(mileageService.getOrgUserMileageSettings).toHaveBeenCalledTimes(1);
+          expect(orgSettingsService.get).toHaveBeenCalledTimes(3);
+          expect(orgUserSettingsService.get).toHaveBeenCalledTimes(2);
+          expect(locationService.getCurrentLocation).toHaveBeenCalledTimes(1);
+          expect(authService.getEou).toHaveBeenCalledTimes(2);
+          expect(component.getMileageByVehicleType).toHaveBeenCalledOnceWith(unfilteredMileageRatesData, 'bicycle');
```
Comment (Dimple16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-2.spec.ts:371]: same for `bicycle`

**Review by Dimple16 — CHANGES_REQUESTED** [review]
one comment

**Review by OmkarJ13 — APPROVED** [review]
Please fix all the requested changes by Dimple first before merging

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/add-edit-mileage/add-edit-mileage-2.spec.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage-2.spec.ts:345]: I could not find any enum for vehicle type. The `PlatformMileageRates` has string type for vehicle

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**